skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import six
+import numpy as np
+from skbio.util._decorator import experimental
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
```

Comment:
Alright, here's a good reason to keep it where it is.

When doing the permutations, I noticed that there is a bit of redundant work (`H` and `XX1` being computed multiple times).  If we were to opt for options 1 or 2, we would be doing _alot_ of repetitive work.  I may do option 3, but that'll only speed up some of the bottlenecks.

So I vote to keep that `regress` function inside of the `mrm`, so that we can optimize more appropriately.  Any objections?